src/content/docs/game-design/godot/2dPlatformerGame.mdx

```diff
@@ -0,0 +1,231 @@
+
+Now open the TileMap menu at the bottom of the screen and click on a tile to enable drawing in the editor. You can click to place individual tiles, or click and drag to paint them following the cursor. Play around with the tile paint tools to create a fun level to play. Some cool things to check out are the Rect tool and the Place Random Tile options.
+
+Now add a new TileMapLayer as a child of Level 1 and place it above Foreground in the scene tree. 
+
+![Screenshot showing scene tree](/src/assets/godot/2DPlatformer/screenshot9.png)
+
+Repeat the steps used for the Foreground (ignoring the physics steps) to create your Background. If you placed the Background TileMapLayer above the Foreground, Background tiles should appear behind the Foreground tiles. Unless your player is below the TileMapLayers, they will disappear behind the background, so make sure to move them below the Foreground in the scene tree.
+
+## Adding Movement
+
+Now we are going to add gravity and movement to the player. Go back to your player scene and attach a script by right clicking the player node and clicking "attach script". Click "create" as the defaults are what we want. Now if you run your game, you should be able to run around. You may want to reduce your speed and jump height, depending on how you want the game to feel (I recommend a speed of 150 and a jump velocity of -300). You may notice that your player doesn't flip when going to the left, nor does the animation change from idle to run when you're moving. To solve this, add the below code above `move_and_slide()` as well as adding `@onready var sprite_node = $AnimatedSprite2D` below `extends CharacterBody2D`. Note that the animation names are case sensitive, so if your player disappears that may be why.
+
+```gdscript
+if velocity.x > 0:
+	sprite_node.flip_h = false
+elif velocity.x < 0:
+	sprite_node.flip_h = true
+
+if abs(velocity.x) > 0:
+	sprite_node.animation = "Run"
+else:
+	sprite_node.animation = "Idle"
+```
+
+## Adding Killzones
+
+Next let's make a killzone so that when you fall off the map or hit an enemy, you'll restart the level. Create a new scene with an Area2D node and rename it Killzone. Now go to Collision in the inspector and change the mask to only have 2 selected. Add a Timer node as a child of Killzone and change the wait time in the inspector to 0 (this will auto change to 0.001). Now let's attach a script to Killzone. Default options are ok. Click on Timer and in the node menu on the right connect the "timeout" signal to the Killzone script. Now click on Killzone and in the node menu, connect the "body entered" signal to the Killzone script. Replace the all of the code in the script with this.
+
+```gdscript
```

Comment:
These should have the filename on them

Reply:
Not sure which file names you're referring to here. If the Killzone's script then I'll need to add a sentence to tell them a specific name for the file